arch/risc-v/src/common/espressif/esp_timer.c

```diff
@@ -539,7 +540,13 @@ int esp_timer_initialize(uint32_t timer_id)
         break;
     }
 
-  snprintf(devpath, sizeof(devpath), "/dev/timer%" PRIu32, timer_id);
+  devpath = lib_get_pathbuffer();
```

Comment:
let us add a configure `CONFIG_LIBC_PATHBUFFER` to enable/disable lib_get_pathbuffer implement.

```
#ifdef CONFIG_LIBC_PATHBUFFER
FAR char *lib_get_pathbuffer(void);
#else
#  define lib_get_pathbuffer() alloca(PATH_MAX)
#endif
```

Reply:
alloca isn't ANSI or POSIX Api, so it isn't available on all possible compiler.

Reply:
Current code base:
https://github.com/apache/nuttx/blob/master/include/string.h#L40-L51
![image](https://github.com/user-attachments/assets/10074721-2305-47a9-a850-b601b23721f9)

I think most toolchains support this feature (even MSVC has a similar implementation)
https://github.com/apache/nuttx/blob/master/include/alloca.h#L30-L42